R/rdeephaven/R/helper_functions.R

```diff
@@ -0,0 +1,121 @@
+################ FUNCTIONS TO GET FROM DH TO R
+
+#' @title Convert a Deephaven TableHandle to an Arrow RecordBatchStreamReader
+#' @description Converts a Deephaven TableHandle to an [Arrow RecordBatchStreamReader](https://arrow.apache.org/docs/r/reference/RecordBatchReader.html),
+#' which can itself be used to produce Arrow tables and R data frames.
+#' 
+#' @param table_handle A Deephaven TableHandle.
+#' @returns An Arrow RecordBatchStreamReader.
+dh_to_rbs_reader <- function(table_handle) {
+
+    ptr <- table_handle$.get_arrowArrayStream_ptr()
+
+    # create R RecordBatchStreamReader from C++ ArrowArrayStream pointed to by ptr
+    rbr <- RecordBatchStreamReader$import_from_c(ptr)
+    return(rbr)
+}
+
+#' @title Convert a Deephaven TableHandle to an Arrow Table.
+#' @description Converts a Deephaven TableHandle to an [Arrow Table](https://arrow.apache.org/docs/r/reference/Table.html),
+#' R data frames.
+#' 
+#' @param table_handle A Deephaven TableHandle.
+#' @returns An Arrow Table.
+dh_to_arrow_table <- function(table_handle) {
+
+    # create R RecordBatchStreamReader
+    rbsr <- dh_to_rbs_reader(table_handle)
+
+    # use RecordBatchStreamReader to get data as arrow table
+    return(rbsr$read_table())
+}
+
+#' @title Convert a Deephaven TableHandle to an R Data Frame.
+#' @description Converts a Deephaven TableHandle to an R Data Frame.
+#' This is the recommended method for converting a Deephaven TableHandle to something that is usable in R.
+#' 
+#' @param table_handle A Deephaven TableHandle.
+#' @returns An R Data Frame.
+dh_to_data_frame <- function(table_handle) {
+
+    arrow_table <- dh_to_arrow_table(table_handle)
+    return(as.data.frame(arrow_table))
+}
+
+################ FUNCTIONS TO GET FROM R TO DH
+
+#' @title Push an Arrow RecordBatchReader to the server as a Deephaven Table.
+#' @description Converts an Arrow RecordBatchReader to a Deephaven Table, pushes that new table to the Deephaven server,
+#' and returns a TableHandle corresponding to the new table.
+#' 
+#' @param client A Deephaven Client object maintaining the connection to the server.
+#' @param rbr An Arrow RecordBatchReader containing the data to be converted to a Deephaven Table.
+#' @param name A string containing the name of the new Deephaven Table to be created on the server.
+#' @returns A Deephaven TableHandle, providing a reference to the new table on the server.
+rbr_to_dh_table <- function(client, rbr, name) {
+
+    if (class(client)[[1]] != "Rcpp_Client") {
```

Comment:
Why is `Rcpp_` in the type name?

Reply:
That's how Rcpp has put it together. I'm just now wondering if I can wrap all of the Rcpp-generated S4 stuff in an R6 class and tailor all the methods and names by hand. That would provide a solution to a lot of what's been discussed in this round of review. The problem with that would be that the underlying Rcpp stuff would likely still be visible at a user level, which sucks.